server/handle_map_capabilities.go

```diff
@@ -75,74 +60,13 @@ func (req HandleMapCapabilities) ServeHTTP(w http.ResponseWriter, r *http.Reques
 		m = m.AddDebugLayers()
 	}
 
-	for i := range m.Layers {
-		// check if the layer already exists in our slice. this can happen if the config
-		// is using the "name" param for a layer to override the providerLayerName
-		var skip bool
-		for j := range tileJSON.VectorLayers {
-			if tileJSON.VectorLayers[j].ID == m.Layers[i].MVTName() {
-				// we need to use the min and max of all layers with this name
-				if tileJSON.VectorLayers[j].MinZoom > m.Layers[i].MinZoom {
-					tileJSON.VectorLayers[j].MinZoom = m.Layers[i].MinZoom
-				}
-
-				if tileJSON.VectorLayers[j].MaxZoom < m.Layers[i].MaxZoom {
-					tileJSON.VectorLayers[j].MaxZoom = m.Layers[i].MaxZoom
-				}
-
-				skip = true
-				break
-			}
-		}
-
-		// the first layer sets the initial min / max otherwise they default to 0/0
-		if len(tileJSON.VectorLayers) == 0 {
-			tileJSON.MinZoom = m.Layers[i].MinZoom
-			tileJSON.MaxZoom = m.Layers[i].MaxZoom
-		}
-
-		// check if we have a min zoom lower then our current min
-		if tileJSON.MinZoom > m.Layers[i].MinZoom {
-			tileJSON.MinZoom = m.Layers[i].MinZoom
-		}
-
-		// check if we have a max zoom higher then our current max
-		if tileJSON.MaxZoom < m.Layers[i].MaxZoom {
-			tileJSON.MaxZoom = m.Layers[i].MaxZoom
-		}
-
-		//	entry for layer already exists. move on
-		if skip {
-			continue
-		}
+	tileJSON := m.GetBaseTileJSON()
 
-		//	build our vector layer details
-		layer := tilejson.VectorLayer{
-			Version: 2,
-			Extent:  4096,
-			ID:      m.Layers[i].MVTName(),
-			Name:    m.Layers[i].MVTName(),
-			MinZoom: m.Layers[i].MinZoom,
-			MaxZoom: m.Layers[i].MaxZoom,
-			Tiles: []string{
-				buildCapabilitiesURL(r, []string{"maps", req.mapName, m.Layers[i].MVTName(), "{z}/{x}/{y}.pbf"}, debugQuery),
-			},
+	//Build tiles urls
+	for i, layer := range tileJSON.VectorLayers {
```

Comment:
I see what you're doing here and why you're doing it, but it has a code smell to me. Part of what I'm hoping to achieve with this PR is positioning us to pass an atlas.Map to the Cache layer so the TileJSON can be easily generated for the MBtiles cache you built. The key issue is the dynamic generation of the HTTP host and URI path prefix. The MBTiles cache provider will need this information but I'm not sure how to best pass it down to that layer without making the interface overly complex. 

I need to think for a bit on how to adjust the Cache interface so we can achieve this while also allowing for dynamic URLS to be generated. If you're open to chatting on this, please join us on Slack: https://slack.go-spatial.org

Reply:
No problem, that why I split the PR to make it more easy to review since some changes are not obvious.
I also think that it was a little "hacky" to do and try to limit it. 
I will try to pass by on slack when I have free time.